src/context/internal.rs

```diff
@@ -205,7 +204,7 @@ extern "C" fn try_catch_glue<'a, 'b: 'a, C, T, F>(rust_thunk: *mut c_void,
     match catch_unwind(AssertUnwindSafe(|| f(cx))) {
         // No Rust panic, no JS exception.
         Ok(Ok(result)) => unsafe {
-            *returned = result.to_raw();
+            *(returned as *mut T) = result;
```

Comment:
This should be using `ptr::write`, since the existing value is garbage and should not be Dropped.